docs/F-Refinements/string-library.md

```diff
@@ -92,53 +100,53 @@ The following program copies the input string only if it contains 20 characters
 
 int main(void)
 {
-        char str[31], copy[21] = "?";
-        int i, len;
-
-        printf("Source : ");
-        scanf("%30[^\n]%*c", str);
-
-        len = strlen(str);
-        
-        if (len < 21)
-        {
-                strcpy(copy, str);
-                printf("Copy : %s\n", copy); 
-        }
-        else 
-        {
-                printf("* No room *\n"); 
-                printf("Copy : %s\n", copy);
-        }
-
-        return 0;
+    char str[31], copy[21] = "?";
```

Comment:
As the array [copy] needs to copy all the letters from the array str, I think the size of the copy array should be the same as str array.

Reply:
That's right! However, in the example, the `strcpy()` works only when the `source` string's length <= `20`. That's why the `destination` string's length is only `21`.